apps/rush/src/actions/CustomRushAction.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as os from 'os';
+import * as colors from 'colors';
+
+import {
+  ICustomOption,
+  ICustomEnumOption,
+  ICustomEnumValue,
+  Stopwatch,
+  Event
+} from '@microsoft/rush-lib';
+
+import {
+  CommandLineFlagParameter,
+  CommandLineIntegerParameter,
+  CommandLineStringListParameter,
+  CommandLineOptionParameter,
+  ICommandLineActionOptions
+} from '@microsoft/ts-command-line';
+
+import RushCommandLineParser from './RushCommandLineParser';
+import { BaseRushAction } from './BaseRushAction';
+import { TaskManager } from '../utilities/TaskManager';
+
+interface ICustomOptionInstance extends ICustomOption {
+  longName: string;
+  parameterValue: CommandLineFlagParameter | CommandLineOptionParameter;
+}
+
+export class CustomRushAction extends BaseRushAction {
+  private customOptions: Array<ICustomOptionInstance> = [];
+
+  private _fromFlag: CommandLineStringListParameter;
```

Comment:
Do all customRushActions support "from" and "to" by default? That is cool.

Reply:
Yep!

---
In reply to: [145582119](https://github.com/Microsoft/web-build-tools/pull/387#discussion_r145582119) [](ancestors = 145582119)